src/twisted/conch/ssh/filetransfer.py

```diff
@@ -527,6 +534,32 @@ def connectionMade(self):
         self.sendPacket(FXP_INIT, data)
 
 
+    def connectionLost(self, reason):
+        """
+        Abort any pending requests.
+        """
+
+        # On Python2, we're an oldstyle class, so we can't use super().
+        FileTransferBase.connectionLost(self, reason)
+        # Not sure why self.connected isn't updated in
+        # t.i.p.Protocol.connectionLost() instead
+        self.connected = False
+
+        # If there are still requests waiting for responses when the
+        # connection is lost, fail them.
+        if self.openRequests:
+            if reason.check(error.ConnectionDone):
+                # Even if our transport was lost "cleanly", our
+                # requests were still not cancelled "cleanly".
+                reqreason = failure.Failure(
```

Comment:
I would prefer to have a better name than `reqreason`... maybe just overwrite/reuse `reason`

And why reuse the message from the `ConnectionDone` failure ...as it is a misleading error.

I think that we should put a custom message here that should better describe what happened and should make it easier for people to track the failure in the code.

Reply:
I've changed this to always use the default message for `ConnectionLost()` but to chain to the original exception with `__cause__`, which I think provides the best of both worlds.